### PR TITLE
[#14] Parallel build and upload jobs

### LINKED ISSUE #14: Parallel build and pushing of Docker images
We have to implement the parallel build of Docker images because a current build takes too long time, even with the enabled cache.

### PR DESCRIPTION
**Pull request by maxkochubey**
Closes #14.

### CODE REVIEW

File: scripts/Jenkinsfile [hunk 1]
```diff
@@ -122,50 +122,45 @@ pipeline {
...
-                        "${buildArgs} --build-arg NOTEBOOK_TYPE=datascience"
-                    )
-                    cicdLibrary.buildDockerImage(
-                        'tensorflow-notebook',
-                        '.',
-                        'containers/jupyter-stacks/Dockerfile',
-                        "${buildArgs} --build-arg NOTEBOOK_TYPE=tensorflow"
-                    )
+                    def buildStages = [:]
+                    ['base', 'datascience', 'tensorflow'].each { type ->
```
Comment (mcdoker18) [inline scripts/Jenkinsfile]: Extract the list to a global variable

File: scripts/Jenkinsfile [hunk 2]
```diff
@@ -103,16 +103,16 @@ pipeline {
...
-                stage("Upload ODAHU PyPi packages") {
+                stage("PyPi packages") {
                     steps {
                         script {
                             cicdLibrary.publishPythonPackage("./", env.param_pypi_private_creds, env.param_pypi_private_url, env.param_pypi_publishing_enabled, env.param_pypi_public_creds, env.param_pypi_public_url)
                         }
                     }
                 }
-                stage('Upload ODAHU NPM packages') {
+                stage('NPM packages') {
```
Comment (karbyshevds) [inline scripts/Jenkinsfile]:
```suggestion
                stage('Publish NPM packages') {
```

File: scripts/Jenkinsfile [hunk 3]
```diff
@@ -103,16 +103,16 @@ pipeline {
             }
         }
 
-        stage('Upload artifacts') {
+        stage('Build ODAHU artifacts') {
             parallel {
-                stage("Upload ODAHU PyPi packages") {
+                stage("PyPi packages") {
```
Comment (karbyshevds) [inline scripts/Jenkinsfile]:
```suggestion
                stage("Publish PyPi packages") {
 ```

File: scripts/Jenkinsfile (hunk 1 above)
Comment (maxkochubey) [reply scripts/Jenkinsfile]: Done in 369bad08822ef06834a803757d1cbb83c556869c.

**Pull request merged**